athena-postgresql/src/main/java/com/amazonaws/athena/connectors/postgresql/PostGreSqlMetadataHandler.java

```diff
@@ -445,4 +446,31 @@ protected PreparedStatement getMaterializedViewOrExternalTable(Connection connec
         LOGGER.debug("Prepared statement for getting name of Materialized View with Case Insensitive Look Up: {}", preparedStatement);
         return preparedStatement;
     }
+
+    /**
+     * Retrieves the names of columns with the data type 'CHAR' for a specified table in a PostgreSQL/Redshift database.
+     *
+     * @param connection the JDBC connection to the database
+     * @param schema Postgresql/Redshift schema name
+     * @param table Postgresql/Redshift table name
+     * @return a list of column names that have the data type 'CHAR'
+     * @throws SQLException if a database access error occurs
+     */
+    public static List<String> getCharColumns(Connection connection, String schema, String table) throws SQLException
+    {
+        List<String> charColumns = new ArrayList<>();
+        String query = "SELECT column_name " +
+                "FROM information_schema.columns " +
+                "WHERE table_schema = ? AND table_name = ? AND data_type = 'character'";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, schema);
+            statement.setString(2, table);
+            try (ResultSet resultSet = statement.executeQuery()) {
```

Comment:
Do we need to make an additional call to the postgresql database to determine this? Is the character datatype gotten from this query different from the datatype we can get from the tableSchema? An apache arrow type has the FieldType for each field in the arrow schema. Is there a case where the FieldType might not align with the type gotten from the postgresql database?

Reply:
Hi @aimethed , yes we make an additional call to postgresql to retreive fixed-length 'char' columns to trim any extra trailing spaces in the data. Yes the datatype obtained from the tableSchema is different which is 'varchar'. Yes it seems apache arrow type does not have a specific data type for fixed-length 'char'.